client/composables/fileTable/useRenameFile.ts

```diff
@@ -0,0 +1,46 @@
+import type { DynamicDialogOptions } from 'primevue/dynamicdialogoptions'
+import type { MessageDTO } from '~/utils/messages'
+import { RenameDialog, RenameDialogFooter } from '#components'
+
+export const useRenameFileDialog = defineStore('renameFileDialog', {
+  state: () => ({ name: '' }),
+})
+
+export default function useRenameFile() {
+  const axios = useAxiosFactory()
+  const { t } = useI18n()
+  const call = useFeedbackCall((fileId: string, newName: string) => {
+    return axios().put<MessageDTO<FileRepresentationDTO>>(`/files/${fileId}/name`, { name: newName })
+  })
+  const { selectedRow, selectedRowId, updateSelectedRow, unselectRow } = useFilesTable()
+  const dialog = useDialog()
+
+  const onClose = (options?: DynamicDialogOptions) => {
```

Comment:
Why?

Reply:
1. Because we did it everywhere
2. Because lambda scope `this` differently than functions, which may cause leakage issues.

That is to say, only make a lambda when it actually doesn't have a name and isn't passed around.